tests/test_integration/test_code_generation.py

```diff
@@ -417,3 +417,32 @@ def stencil(
 
     outp.device_to_host()
     assert np.allclose(1.0, np.asarray(outp))
+
+
+# TODO(jdahm): Add additional backend support and move to test_suites
+@pytest.mark.parametrize(
+    "backend", GTC_GRIDTOOLS_BACKENDS + [pytest.param("numpy"), pytest.param("debug")]
+)
+def test_write_data_dim_indirect_addressing(backend):
+    @gtscript.stencil(backend=backend)
+    def stencil(
+        input_field: gtscript.Field[gtscript.IJK, np.float_],
+        output_field: gtscript.Field[gtscript.IJK, (np.float_, (1,))],
+        index: int,
+    ):
+        with computation(PARALLEL), interval(...):
+            output_field[0, 0, 0][index] = input_field
```

Comment:
Since the index is an expression let's add a test for `output_field[0, 0, 0][index+1]`. Something like `output_field[0, 0, 0][index+1.1]` will probably also pass the frontend and might give a terrible error message somewhere, but I think this is fine.

Reply:
I check this in a unit test in `test_gtir.py` now. Is that sufficient?

Reply:
Yes, should be fine.